rcsb/app/file/uploadRequest.py

```diff
@@ -90,217 +118,33 @@ async def upload(
-    if contents and len(contents) <= 0:
-        raise HTTPException(status_code=400, detail="error - empty file")
-    try:
-        # save, then hash, then decompress
-        # should lock, however client must wait for each response before sending next chunk, precluding race conditions (unless multifile upload problem)
-        async with aiofiles.open(tempPath, "ab") as ofh:
-            await ofh.seek(chunkOffset)
-            await ofh.write(contents)
-            await ofh.flush()
-            os.fsync(ofh.fileno())
-        # if last chunk
-        if chunkIndex + 1 == expectedChunks:
-            if hashDigest and hashType:
-                if hashType == "SHA1":
-                    hashObj = hashlib.sha1()
-                elif hashType == "SHA256":
-                    hashObj = hashlib.sha256()
-                elif hashType == "MD5":
-                    hashObj = hashlib.md5()
-                blockSize = 65536
-                # hash temp file
-                with open(tempPath, "rb") as r:
-                    while hash_chunk := r.read(blockSize):
-                        hashObj.update(hash_chunk)
-                hexdigest = hashObj.hexdigest()
-                ok = (hexdigest == hashDigest)
-                if not ok:
-                    raise HTTPException(status_code=400, detail=f"{hashType} hash check failed")
-                else:
-                    # lock then save
-                    lockPath = os.path.join(os.path.dirname(filePath), "." + os.path.basename(filePath) + ".lock")
-                    lock = FileLock(lockPath)
-                    try:
-                        with lock.acquire(timeout=60 * 60 * 4):
-                            # last minute race condition handling
-                            if os.path.exists(filePath) and not allowOverwrite:
-                                raise HTTPException(status_code=400, detail="Encountered existing file - cannot overwrite")
-                            else:
-                                # save final version
-                                os.replace(tempPath, filePath)
-                    except Timeout:
-                        raise HTTPException(status_code=400, detail=f"error - lock timed out on {filePath}")
-                    finally:
-                        lock.release()
-                        if os.path.exists(lockPath):
-                            os.unlink(lockPath)
-                # remove temp file
-                if os.path.exists(tempPath):
-                    os.unlink(tempPath)
-                # decompress
-                if decompress:
-                    with gzip.open(filePath, "rb") as r:
-                        with open(tempPath, "wb") as w:
-                            w.write(r.read())
-                    os.replace(tempPath, filePath)
-            else:
-                raise HTTPException(status_code=500, detail="Error - missing hash")
```

Comment:
Is there a reason this isn't needed anymore? How will non-resumable uploads be handled?

Reply:
Explanation above.